steps.go

```diff
@@ -697,26 +700,38 @@ func (tc *TestContext) cPUMetricsOfAllInstancesInTheInstancePoolMustBeVisibleInP
 			tc.logger.Debugf("failed to get instance pool (%v)", data.Data.ResultType)
 			continue
 		}
+
+		var instancePoolIps []string
+		var metricIpsSliced []string
 		metricIps := map[string]bool{}
 		for _, record := range data.Data.Result {
 			parts := strings.SplitN(record.Metric.Instance, ":", 2)
 			metricIps[parts[0]] = true
 		}
+		for k := range metricIps {
+			metricIpsSliced = append(metricIpsSliced, k)
+		}
 		allIpsFound := true
 		for _, vm := range instancePool.VirtualMachines {
 			for _, nic := range vm.Nic {
-				if nic.IsDefault {
+				if nic.IsDefault && nic.IPAddress != nil {
+					instancePoolIps = append(instancePoolIps, nic.IPAddress.String())
```

Comment:
### Reason
The default nic (`nic.IsDefault == true`) was sometimes `nil`, which lead to the test failing, even though only one (non-instpool) vm was present.

### Remark
While this could possibly be due to erroneous implementations, these (default nic)IPs/VMs were transient, and never showed up as additional/multiple instances e.g. within the exoscale portal.
(Possibly delayed/asynchronous shutting down of VMs, used in previous tests?)